### ISSUE TITLE
Static expressions don't work with variables on other side of ternary

### ISSUE THREAD

**Issue by joelpittet** [twigphp/Twig#2146]
Here's an example. I'll try to debug, but I guess the expression needs to be evaluated separately.
http://twigfiddle.com/xo63vc/3

Here's the generated code for that.

```
  escaped: {{ name }}
  static: {{ '&' }}
  true: {{ true ? '&' : '&' }}
  false: {{ false ? '&' : '&' }}
  true with variable false: {{ true ? '&' : name }}
  false with variable true: {{ false ? name : '&' }}

  true with variable false: {{ true ? '&'|raw : name }}
  false with variable true: {{ false ? name : '&'|raw }}
```

```
    protected function doDisplay(array $context, array $blocks = array())
    {
        // line 1
        echo "  escaped: ";
        echo twig_escape_filter($this->env, (isset($context["name"]) ? $context["name"] : null), "html", null, true);
        echo "
  static: ";
        // line 2
        echo "&";
        echo "
  true: ";
        // line 3
        echo ((true) ? ("&") : ("&"));
        echo "
  false: ";
        // line 4
        echo ((false) ? ("&") : ("&"));
        echo "
  true with variable false: ";
        // line 5
        echo twig_escape_filter($this->env, ((true) ? ("&") : ((isset($context["name"]) ? $context["name"] : null))), "html", null, true);
        echo "
  false with variable true: ";
        // line 6
        echo twig_escape_filter($this->env, ((false) ? ((isset($context["name"]) ? $context["name"] : null)) : ("&")), "html", null, true);
    }
```

**Comment by fabpot** [comment]
see #449

**Comment by joelpittet** [comment]
Thanks I'm still wondering if we can split the `twig_escape_filter` into two and do the safe analysis around each side of the `:` expression?  May not be possible, though thanks for the reference, my search of the queue didn't yield anything.  At the least we could put a caveat in the docs.

**Comment by fabpot** [comment]
Will be fixed by #2934
